Sources/AppcuesKit/Push/PushMonitor.swift

```diff
@@ -0,0 +1,132 @@
+    func didReceiveNotification(userInfo: [AnyHashable: Any], completionHandler: @escaping () -> Void) -> Bool
+}
+
+internal class PushMonitor: PushMonitoring {
+
+    private weak var appcues: Appcues?
+    private let storage: DataStoring
+
+    private var pushAuthorizationStatus: UNAuthorizationStatus = .notDetermined
+
+    var pushEnabled: Bool {
+        pushAuthorizationStatus == .authorized && storage.pushToken != nil
+    }
+
+    var pushBackgroundEnabled: Bool {
+        storage.pushToken != nil
+    }
+
+    var pushPrimerEligible: Bool {
+        pushAuthorizationStatus == .notDetermined
+    }
+
+    init(container: DIContainer) {
+        self.appcues = container.owner
+        self.storage = container.resolve(DataStoring.self)
+
+        refreshPushStatus()
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(applicationWillEnterForeground),
+            name: UIApplication.willEnterForegroundNotification,
+            object: nil
+        )
+    }
+
+    @objc
+    private func applicationWillEnterForeground(notification: Notification) {
+        refreshPushStatus()
+    }
+
+    func refreshPushStatus(completion: ((UNAuthorizationStatus) -> Void)? = nil) {
+        // Skip call to UNUserNotificationCenter.current() in tests to avoid crashing in package tests
+        #if DEBUG
+        guard ProcessInfo.processInfo.environment["XCTestBundlePath"] == nil else {
+            completion?(pushAuthorizationStatus)
+            return
+        }
+        #endif
+
+        UNUserNotificationCenter.current().getNotificationSettings { [weak self] settings in
+            self?.pushAuthorizationStatus = settings.authorizationStatus
+            completion?(settings.authorizationStatus)
+        }
+    }
+
+    // `completionHandler` should be called iff the function returns true.
+    func didReceiveNotification(userInfo: [AnyHashable: Any], completionHandler: @escaping () -> Void) -> Bool {
+        guard let parsedNotification = ParsedNotification(userInfo: userInfo) else {
+            // Not an Appcues push
+            return false
+        }
+
+        guard let appcues = appcues else {
+            return false
+        }
+
+        // If there's a user ID mismatch, don't do anything with the notification
+        guard parsedNotification.userID == storage.userID else {
+            completionHandler()
+            return true
+        }
+
+        // If no session, start one for the user in the notification
+        if !appcues.isActive {
+            storage.userID = parsedNotification.userID
+            storage.isAnonymous = false
+        }
+
+        let analyticsPublisher = appcues.container.resolve(AnalyticsPublishing.self)
+        analyticsPublisher.publish(TrackingUpdate(
+            type: .event(name: Events.Push.pushOpened.rawValue, interactive: false),
+            properties: [
+                "notification_id": parsedNotification.notificationID
```

Comment:
lets add `workflow_id`, `workflow_task_id` and `transaction_id` at least to start here

there are some push notification events defined here https://www.notion.so/appcues/Workflow-Analytics-Event-Schema-48b243655c394446b665ac8f759476fd?pvs=4 (scroll down a bit)

the other items should already be in there by default - account, user, device

Reply:
Do we want to send the device auto props with `appcues:push_opened`? That's the only way `_deviceId` would make it into the payload.

Reply:
the `_localId` in auto props is technically the same as `_deviceId`, but maybe yeah just add `device_id` right here as well to make it very clear since it's a key thing for this particular analytic